src/modules/esl-utils/async/promise/event.ts

```diff
@@ -33,6 +33,9 @@ export function promisifyEvent(
     if (typeof timeout === 'number' && timeout >= 0) {
       setTimeout(eventCallback, timeout);
     }
+    if (typeof options === 'object') {
+      options.signal?.addEventListener('abort', () => reject(new Error('Rejected by abort signal')), {once: true});
+    }
```

Comment:
Memory leak: we have 2 cases when we end with signal subscription:
1) Negative (when original event observation is canceled by signal) - in that case we have an `abort` event and once will cover our requirement
2) Positive (the original Event was caught, so no signal involvement before event) - and here we end up with useless signal object that can not be garbage-collected as it have an active subscribtion.